app/views/layouts/_market_nav.html.erb

```diff
@@ -0,0 +1,5 @@
+<%= link_to 'All Markets', markets_path %>
+
+<%= link_to 'Market', market_path(@market) if params[:market_id] %>
+
+<%= link_to 'Logout', :root %>
```

Comment:
You should move this kind of partial into the folder it's related to. Or, if it's a partial that will be shared across folders, you could stick it in a shared folder.